test/widgets/text_test.dart

```diff
@@ -0,0 +1,111 @@
+import 'dart:ui';
+
+import 'package:checks/checks.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:zulip/widgets/text.dart';
+
+void main() {
+  group('weightVariableTextStyle', () {
+    Future<void> runCheck(
+      String description,
+      {
+        required TextStyle Function(BuildContext context) styleBuilder,
```

Comment:
nit:
```suggestion
    Future<void> runCheck(
      String description, {
      required TextStyle Function(BuildContext context) styleBuilder,
```

For some examples in the Flutter repo, see the `Text` constructors.